src/types.rs

```diff
@@ -159,6 +195,46 @@ pub struct Block {
     pub mix_hash: Option<String>,
 }
 
+impl ToChecksummed for Block {
+    fn to_checksummed(self) -> Result<Self> {
+        Ok(Block {
+            miner: self.miner.to_checksummed().context("miner")?,
+            ..self
+        })
+    }
+}
+
+impl<T: ToChecksummed> ToChecksummed for Option<T> {
+    fn to_checksummed(self) -> Result<Self> {
+        Ok(match self {
+            Some(v) => Some(v.to_checksummed()?),
+            None => None,
+        })
+    }
+}
+
+impl ToChecksummed for String {
+    fn to_checksummed(self) -> Result<Self> {
+        Ok(alloy_primitives::Address::from_hex(self)
+            .context("creating address from hex string")?
+            .to_checksum(None))
+    }
+}
```

Comment:
@ozgrakkurt these two impl are essentially my reasoning for using a trait here. It allows me to call the method directly on a string, and on an option type of any other type that implements the trait. It's pretty clean way of chaining these calls.

Reply:
can do `encode_addr(addr: &[u8], checksum: bool)`. Don't think we need to call it on strings. They are all byte arrays afaik. If need to call on string then can just do another function since don't think it will change the calling code that much